### ISSUE TITLE
Paladin's Vengeance +15% damage boost to Consecration should be dynamic not static as it is currently

### ISSUE THREAD

**Issue by theloras** [LightsHope/issues#12]
OK, let's try this again, shall we?

Basically what is currently happening is that if I cast Consecration and start getting it's normal tick dmg but then get a crit during the 8 second duration, it will still only do it's normal damage and not gain the benefit of +15% Holy dmg to each tick.

How it should work is that if I drop Consecration and get normal damage ticks, then get a crit mid-duration, the consecration ticks should gain the +15% damage boost dynamically as is the case when adding Judgement of the Crusader mid Consecration duration.

JotC works properly in this case whereas Vengeance does not - aka if I drop Consecration without JotC up and get normal Consecration ticks but judge crusader mid 8 sec, Consecration's damage does dynamically gain the benefit of +Holy dmg to the target.

Seal of Righteousness + Judgement of Righteousness, Seal of Command + Judgement of Command and all of our Undead damage abilities dynamically gain the benefit from Vengeance but Consecration does not.

**Rank 14 Critadin ftw**
https://youtu.be/jPgv6fmVpLs?t=246

Mobywan has Vengeance active when he drops his Consecration and you see it tick for 64 Holy damage - once his Vengeance wears off however, his Consecration ticks go down to only 55 Holy damage.

Even though the Youtube video was uploaded in 2014 - Mobywan actually first published it back during retail Vanilla as can be seen in his warcraftmovies.com page:

**R14 Critadin ftw by Mobywan**
http://www.warcraftmovies.com/movieview.php?id=32000
2006-11-27

**Therrik Paladin PvP**
_Published on Jan 3, 2007_
https://www.youtube.com/watch?v=v89yIkiVN6Y&feature=youtu.be&t=84

Even though this is right after the 2.0 pre-patch for TBC, you can see that the same mechanics applied - Therik drops his Consecration which does 62 Holy damage but gets a crit mid duration, gains Vengeance which makes his Cons damage go up to 71 Holy damage.

TLDR - Consecration's damage output should update dynamically not only from debuffs on the target (currently working) but also from buffs on the Paladin as well (currently not working).

If the Paladin gains Vengeance or pops a spell damage trinket for example - then Consecration's damage should go up on the fly as a result - conversely if the Paladin loses Vengeance or their spell damage trinket buff fades, then Consecration should likewise go down in damage on the fly.

here is an example from retail vanilla showing BOTH Vengeance and ZHC trinket being popped and Consecration's damage being updated dynamically:

**Xolkar 20/0/31 Paladin PvP**
_Published on Aug 13, 2006_
https://youtu.be/Pk_kHHI0t4o?t=40

Xolkar pops his ZHC and has Vengeance as he drops Consecration which does 80-82 Holy dmg ticks but once his Vengeance fades, it drops down to 70.

Then again in the very next sequence Xolkar drops Consecration with its first initial ticks at 68 dmg but then gets Vengeance and pops his ZHC trinket which boost each tick to 89 dmg:

https://youtu.be/Pk_kHHI0t4o?t=56

**Heero 2 - Paladin PvP (1 of 2)**
https://youtu.be/f6SPpRZmuBU?t=297

was first uploaded on 2006-12-06 to http://www.warcraftmovies.com/movieview.php?id=32621

Heero drops Consecration as he pops ZHC and gains Vengeance and its damage does 92 per tick - a few seconds later he loses Vengeance buts still has ZHC left but Consecration's damage has gone down to 78-79 ticks

**Retribution Paladin - Timangi 1 PvP - Cromfel's Timangi**
https://youtu.be/peIxTVTdHNs?t=254

was first uploaded on 2006-08-12 to http://www.warcraftmovies.com/movieview.php?id=25814

Timangi drops rank 1 Consecration which starts out doing 23 dmg ticks but gains Vengeance mid duration and it goes up to 26 dmg ticks.

Happens again at the 5:20 mark - https://youtu.be/peIxTVTdHNs?t=320

and at 5:38 mark - https://youtu.be/peIxTVTdHNs?t=338

again at 10:48 - https://youtu.be/peIxTVTdHNs?t=648

**Retribution Paladin - Timangi 2 PvP - Scholae Palatinae**
https://youtu.be/2HMJNP1PbZM?t=162

was first uploaded on 2006-11-27 to http://www.warcraftmovies.com/movieview.php?id=32032

Timangi drops Consecration which ticks for 45 damage, gains Vengeance which then boosts Consecration ticks to 50 dmg.

again at 4:23 - https://youtu.be/2HMJNP1PbZM?t=263 - Cons with no Venge ticks at 32 and with Venge ticks at 37

perfect example of dynamic buff and loss at 16:05 - https://youtu.be/2HMJNP1PbZM?t=965 - Timangi drops Cons and it ticks for for 40 dmg, gets a crit and gains Vengeance to boost ticks to 45 dmg then Shaman purges Vengeance and ticks drop back down to 40

**Orrah 2 Paladin PvP (2 of 2)**
https://youtu.be/Gux0AquyU7M?t=134

was first uploaded on 2006-06-14 to http://www.warcraftmovies.com/movieview.php?id=22225

Orrah pops both ZHC and has Vengeance when he drops Consecration which ticks for 73-74 Holy dmg then ZHC fades (but refreshes Vengeance) and his Consecration ticks drop down to 62

**Comment by theloras** [comment]
any word on when this will make it to live server?

**Comment by theloras** [comment]
this was fixed with this week's patch = happy in the pants
